src/Admin/AbstractAdmin.php

```diff
@@ -851,7 +851,7 @@ public function buildDatagrid()
             return;
         }
 
-        $filterParameters = $this->getFilterParameters();
+        $filterParameters = array_merge($this->getFilterParameters(), $this->determineSorting());
```

Comment:
Currently, the $parameters are computed this way
```
$parameters = array_merge(
                $this->getModelManager()->getDefaultSortValues($this->getClass()),
                $this->datagridValues, // NEXT_MAJOR: Remove this line.
                $this->getDefaultSortValues(),
                $this->getDefaultFilterValues(),
                $filters
            );
```

It feels weird to me, to compute this and to re-compute something else, and then to merge it. It will easily introduce bugs.